README.md

```diff
@@ -49,3 +49,31 @@ tree.Sorter = func(u1, u2 fyne.URI) bool {
 <p align="center" markdown="1" style="max-width: 100%">
   <img src="img/widget-filetree.png" width="1024" height="880" alt="FileTree Widget" style="max-width: 100%" />
 </p>
+
+### CompletionEntry
+
+An extension of widget.Entry for displaying a popup menu for completion. Keyboard keys can be used to navigate (up and down) and select the option to set to the Entry (Enter key) - the mouse can be used too.
+
+```go
+entry := widget.NewCompletionEntry([]string{})
+entry.OnChanged = func(s string) {
+    if len(s) < 3 {
+        // for example, you can avoid to display the completion
+        // if the entry has a value lenght < 3
+        entry.HideCompletion()
+        return
+    }
+
+    // here, you can get compltion from a dataset then
+    // set the list to the completion
+    entry.SetOptions([]string{"foo", "bar", "baz"})
```

Comment:
I don’t know if it is just me, but it seems odd to create the entry with no items at all and then add them in the OnChanged function.

Reply:
That was to simplify the example, but if you want I can change the example.

Reply:
Finally, I don't understand the problem you point here. The completion is commonly dynamically changed while the user type text. So it's normal to have an empty completion and to change it on change.

Here, the completion is static, so that could be badly understood. But keep in mind an example of completing the word from an API or database while user type.

I can probably give more explanation in the documentation.

Reply:
Alright. I might have misunderstood your meaning with this. I thought it just was supposed to auto complete from a given set of options passed to it.

Reply:
So yes, my example is bad :) I will provide another one that is more explicit.

Reply:
I changed the example with a completion on Wikipedia API (and set the screenshot corresponding to  the result)